modules/core/shared/src/main/scala/lucuma/core/math/BoundedInterval.scala

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.math
+
+import cats.Eq
+import cats.Order
+import cats.syntax.all.*
+import spire.math.Bounded
+import spire.math.Empty
+import spire.math.Interval
+import spire.math.Point
+import spire.math.interval.ValueBound
+
+/**
+  * A `BoundedInterval` is a `spire.math.Interval` but with both bounds defined.
+  * Therefore, it can only be a `Bounded` or `Point`. It simplifies usage,
+  * without having to worry about the `Empty`, `All`, `Above` and `Below` cases.
+  */
+type BoundedInterval[A] = Bounded[A] | Point[A]
+
+object BoundedInterval:
+  def closed[A: Order](lower: A, upper: A): Option[BoundedInterval[A]] = 
+    Interval.closed(lower, upper).some.filterNot(_.isEmpty).map(_.asInstanceOf[BoundedInterval[A]])
+
+  def unsafeClosed[A: Order](lower: A, upper: A): BoundedInterval[A] = 
+    closed(lower, upper).getOrElse(sys.error(s"Could not build a bounded closed interval with lower bound [$lower] and upper bound [$upper]."))
+
+  def openUpper[A: Order](lower: A, upper: A): Option[BoundedInterval[A]] = 
+    Interval.openUpper(lower, upper).some.filterNot(_.isEmpty).map(_.asInstanceOf[BoundedInterval[A]])
```

Comment:
Are `closed` and `openUpper` the only cases we're interested in?

Reply:
Those are the only ones we are using for the moment, but I just added the other ones for completeness.